src/main/java/address/storage/StorageManager.java

```diff
@@ -181,16 +181,20 @@ public static UserPrefs getUserPrefs(File prefsFile) {
     public void start() {
         logger.info("Starting storage manager.");
         loadDataFromFile(prefs.getSaveLocation());
-        raise(new SaveLocationChangedEvent(prefs.getSaveLocation()));
```

Comment:
Initializing the label should not be done using this event. We should not fire the event for the sole purpose of causing side effects. The label should be initialized simply by getting the value from model manager.